mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala

```diff
@@ -87,6 +87,14 @@ final class DecisionTreeRegressor @Since("1.4.0") (@Since("1.4.0") override val
     trees.head.asInstanceOf[DecisionTreeRegressionModel]
   }
 
+  /** (private[ml]) Train a decision tree on an RDD */
+  private[ml] def train(data: RDD[LabeledPoint],
```

Comment:
This seems fine for now as it is `private`. The main idea with this PR will be to start the migration. As part of the remaining steps in [SPARK-12326](https://issues.apache.org/jira/browse/SPARK-12326), there should be plenty of opportunity to clean things up.